### PR TITLE
Extend kubernetes client flags to match kubectl

### LINKED ISSUE #1810: k8s client forgets to use CA when API URL specified
One user found that due to their in-cluster networking, they had to specify `--probe.kubernetes.api=https://loadbalanced-k8s-api.mycorp.com:8443`. When this was enabled, the k8s client in the scope probe forgot about the CA cert that was available in the pod secrets, and errored out with `x509: certificate signed by unknown authority`.

Let's find a way to allow the user to specify that they should use the in-pod secret with an override on the API URL.

cc @2opremio @errordeveloper

### PR DESCRIPTION
**Pull request by 2opremio**
Fixes #1810

### CODE REVIEW

**Comment by 2opremio** [comment]
@mindfulmonk would you mind testing this to see if it solves your problem? You can use the Docker image `weaveworks/scope:1810-extend-k8s-client-flags-1bdff95`

**Comment by mindfulmonk** [comment]
@2opremio Not a problem, just waiting for it to appear in https://hub.docker.com/r/weaveworks/scope/tags/ ;)

**Comment by 2opremio** [comment]
> @2opremio Not a problem, just waiting for it to appear in https://hub.docker.com/r/weaveworks/scope/tags/ ;)

Ops, please try again, it should be there now

**Comment by 2opremio** [comment]
Circle is incorrectly bitching about spelling mistakes.

File: probe/kubernetes/client.go [hunk 1]
```diff
@@ -62,35 +64,78 @@ func runReflectorUntil(r *cache.Reflector, resyncPeriod time.Duration, stopCh <-
...
 // NewClient returns a usable Client. Don't forget to Stop it.
-func NewClient(addr string, resyncPeriod time.Duration) (Client, error) {
-	var config *restclient.Config
-	if addr != "" {
-		config = &restclient.Config{Host: addr}
-	} else {
-		// If no API server address was provided, assume we are running
+func NewClient(config ClientConfig) (Client, error) {
+	var restConfig *restclient.Config
+	if config.Server == "" && config.Kubeconfig == "" {
```
Comment (paulbellamy) [inline probe/kubernetes/client.go]: Why do I need server *and* kubeconfig? I thought kubeconfig included the server?

**Comment by paulbellamy** [comment]
If you don't provide a kubernetes context, will it use the current context?

**Comment by 2opremio** [comment]
> If you don't provide a kubernetes context, will it use the current context?

Yes, just like kubectl (that's what my experimentation shows with our service kubeconfig files).

File: probe/kubernetes/client.go [hunk 2]
```diff
@@ -62,35 +64,79 @@ func runReflectorUntil(r *cache.Reflector, resyncPeriod time.Duration, stopCh <-
...
+				},
+				CurrentContext: config.Context,
+			},
+		).ClientConfig()
+		if err != nil {
 			return nil, err
 		}
+
 	}
+	log.Infof("kubernetes: targetting api server %s", restConfig.Host)
```
Comment (jml) [inline probe/kubernetes/client.go]: Should be "targeting".

**Comment by jml** [comment]
LGTM, modulo spelling mistake.

**Pull request merged**